tests/tools/test_ngsderive.yml

```diff
@@ -0,0 +1,55 @@
+- name: infer_strandedness_miniwdl
+  tags:
+    - miniwdl
+    - ngsderive
+  command: >-
+    miniwdl run -d test-output/. --copy-input-files --task infer_strandedness tools/ngsderive.wdl bam="tests/tools/input/test.bwa_aln_pe.bam" bam_index="tests/tools/input/test.bwa_aln_pe.bam.bai" gtf="tests/tools/input/genes.gtf.gz"
+  files:
+    - path: test-output/out/strandedness_file/test.bwa_aln_pe.strandedness.txt
+      contains:
+        - "Unstranded"
+
+- name: instrument_miniwdl
+  tags:
+    - miniwdl
+    - ngsderive
+  command: >-
+    miniwdl run -d test-output/. --copy-input-files --task instrument tools/ngsderive.wdl bam="tests/tools/input/test.bwa_aln_pe.bam"
+  files:
+    - path: test-output/out/instrument_file/test.bwa_aln_pe.instrument.txt
+      contains:
+        - "multiple instruments"
+        - "unknown confidence"
+
+- name: read_length_miniwdl
+  tags:
+    - miniwdl
+    - ngsderive
+  command: >-
+    miniwdl run -d test-output/. --copy-input-files --task read_length tools/ngsderive.wdl bam="tests/tools/input/test.bwa_aln_pe.bam" bam_index="tests/tools/input/test.bwa_aln_pe.bam.bai"
+  files:
+    - path: test-output/out/read_length_file/test.bwa_aln_pe.readlength.txt
+      contains:
+        - "150=20000"
+
+- name: encoding_miniwdl
+  tags:
+    - miniwdl
+    - ngsderive
+  command: >-
+    miniwdl run -d test-output/. --copy-input-files --task encoding -i tests/tools/input_json/ngsderive_encoding.json tools/ngsderive.wdl
+  files:
+    - path: test-output/out/encoding_file/test.bwa_aln_pe.encoding.txt
+      contains:
+        - "ASCII range: 74-74"
```

Comment:
The only PHRED encoding present here is `74`? Where did this test BAM come from?

Reply:
It's synthetic data from `ngs generate`. Apparently, it doesn't simulate quality scores realistically by default.

Reply:
That's fine then. I think this is the first time `encoding` has flagged something other than PHRED+33 encoding (that I didn't synthesize for testing). Nice to see it catching something "real" for once!